### ISSUE TITLE
[Feature Request] Add a module settings grouping mechanism

### ISSUE THREAD

**Issue by p4535992** [foundryvtt/foundryvtt#9049]
### User Experience

If you are like me , you really have very little time to devote to free module development, one thing I like less and less is to prepare a long list of module settings and see them displayed in the classic "list"  view low unintuitive because often the settings at the level of logic should be grouped in appropriate groups.

Yes I know I could make my own custom dialog settings and handle everything the way I want it like I did with Tidy5e, the problem is always time...

A simple solution I came up with is the use of html "fieldsets". Adding two additional fields "group" and "subgroup" to indicate a "fieldset" structure and add that intutiveness that FoundryVTT's default settings mechanism lacks.

Here a example of what i have in mind:

```
	game.settings.register(CONSTANTS.MODULE_NAME, "XXX", {
		name: "XXX",
		scope: "client",
		type: String,
		default: "XXX",
		config: true,
		onChange: refresh,
                group: "My Name FieldSet 1 (is the <legend> text element to uderstand each other)",
                subgroup: "My Name FieldSet 2 under FieldSet 1 (is the <legend> text element to uderstand each other)"
	});
```

**Comment by mclemente** [comment]
I was about to open a similar request. At times it feels like I'm developing a settings menu instead of developing a module due to how cumbersome it is to make the module more customizable.

I'd really enjoy if there was a simple way to populate settings menus. There's no way to add settings to it besides making a non-config setting and then add it through to the html plus its own FormApplication.

In a similar vein to groups/subgroups, I'd wish just naming the target menu would add them without the overhead of adding them to a FormApplication. Also, the subgroup suggestion could be used for setting tabs, similar to Core's Audio/Video menu, maybe we could set the tabs when calling `game.settings.registerMenu()`?

**Comment by caewok** [comment]
Related: #8602

**Comment by aaclayton** [comment]
Merged into #8602
